ui/src/app/settings/components/SettingsTable/SettingsTable.tsx

```diff
@@ -0,0 +1,187 @@
+import type { HTMLProps, ReactNode } from "react";
+
+import {
+  Button,
+  Link as VanillaLink,
+  MainTable,
+  SearchBox,
+  Spinner,
+  Tooltip,
+} from "@canonical/react-components";
+import classNames from "classnames";
+import { Link } from "react-router-dom";
+
+// TODO: these should eventually use the react-components types
+// when they have been migrated to TypeScript.
+type MainTableHeader = {
```

Comment:
These types are copied from react-components because we haven't done the upgrade yet. I also found a few issues with the types so I'll create a PR for react-components. I'll file an issue to remove these types once the new types are released.

Reply:
Here's the PR: https://github.com/canonical-web-and-design/react-components/pull/542.